hudi-common/src/main/java/org/apache/hudi/common/table/HoodieTableConfig.java

```diff
@@ -298,8 +298,9 @@ public String getBootstrapIndexClass() {
   }
 
   public static String getDefaultBootstrapIndexClass(Properties props) {
+    HoodieConfig hoodieConfig = new HoodieConfig(props);
     String defaultClass = BOOTSTRAP_INDEX_CLASS_NAME.defaultValue();
-    if ("false".equalsIgnoreCase(props.getProperty(BOOTSTRAP_INDEX_ENABLE.key()))) {
+    if (!hoodieConfig.getBooleanOrDefault(BOOTSTRAP_INDEX_ENABLE)) {
```

Comment:
The option `hoodie.bootstrap.index.class` could not have the default value. If the default value of the `hoodie.bootstrap.index.class` is `HFileBootstrapIndex.class.getName()`, the method `getDefaultBootstrapIndexClass` should be renamed.